apps/programmes/views.py

```diff
@@ -15,3 +18,21 @@ class ProgrammeViewSet(GenericViewSet, ListModelMixin, RetrieveModelMixin):
 
     authentication_classes = [JWTAuthentication]
     permission_classes = [IsAuthenticated]
+
+class ProgrammeStructureView(APIView):
+    authentication_classes = [JWTAuthentication]
+    permission_classes = [IsAuthenticated]
+
+    def get(self, request, programme_id):
+        try:
+            programme = Programme.objects.get(id=programme_id)
+            serializer = ProgrammeSerializer(programme)
+            programme_data = serializer.data
+
+            programme_courses = ProgrammeCourse.objects.filter(programme=programme)
+            course_data = ProgrammeCourseSerializer(programme_courses, many=True).data
```

Comment:
why not use CourseSerializer?

Reply:
CourseSerializer would give a more general view of the course, lacking the context that ProgrammeCourseSerializer provides with programme-specific prerequisite handling.
CourseSerializer serializes the generic Course data without any programme context

Reply:
but programmeCourse is just returning Course info